### ISSUE TITLE
tests/acl/test_acl.py needs to pick correct set of interfaces

### ISSUE THREAD

**Issue by rraghav-cisco** [Azure/sonic-mgmt#3158]
Currently, the script test/acl/test_acl.py is picking a set of 8 interfaces and shutting the rest down using an algorithm as below:

'''
    topo = tbinfo["topo"]["type"]
    for interface, neighbor in mg_facts["minigraph_neighbors"].items():
        port_id = mg_facts["minigraph_ptf_indices"][interface]
        if (topo == "t1" and "T0" in neighbor["name"]) or (topo == "t0" and "Server" in neighbor["name"]):
            downstream_ports.append(interface)
            downstream_port_ids.append(port_id)
        elif (topo == "t1" and "T2" in neighbor["name"]) or (topo == "t0" and "T1" in neighbor["name"]):
            upstream_ports.append(interface)
            upstream_port_ids.append(port_id)
'''

The above code works on the assumption that DATA_INGRESS_ACL is applied on all interfaces in the system. Unfortunately, this might not always be the case.

The problem with is approach is, that if the user had some ACL already set on some of the interfaces, the above algorithm might end up picking interfaces on which there is no ACL at all. We should also consider picking up interfaces which are part of the ACL.

**Comment by yxieca** [comment]
@rraghav-cisco can you clarify the scenario a bit more?

Is the test still fine if DATA_INGRESS_ACL  is applied on all interfaces?

**Comment by rraghav-cisco** [comment]
Yes, if the ACL is applied on all interfaces, we are fine. If not, the script needs to pick only the interfaces where the ACL is applied.

**Comment by yxieca** [comment]
@rraghav-cisco Thanks for the clarification. Do you think Cisco can help update the test?

**Comment by rraghav-cisco** [comment]
sure. We had internally modified it to apply and work with only interfaces that have ACLs on them, and shutdown interfaces that don't have ACLs. We will set a PR to the community soon.

**Comment by arista-nwolfe** [comment]
I'm seeing this behavior on single asic linecards where Ethernet ports don't have the ACLs applied to them.
After applying the following diff I saw this issue resolved and Ethernet ports had ACLs applied to them.

git diff test_acl_orig.py test_acl_new.py
diff --git a/test_acl_orig.py b/test_acl_new.py
index 1bd1fef..86d6548 100644
--- a/test_acl_orig.py
+++ b/test_acl_new.py
@@ -146,11 +146,11 @@ def get_t2_info(duthosts, tbinfo):
             for interface, neighbor in mg_facts["minigraph_neighbors"].items():
                 port_id = mg_facts["minigraph_ptf_indices"][interface]
                 if "T1" in neighbor["name"]:
-                    downstream_ports_per_dut[neighbor['namespace']].append(interface)
+                    downstream_ports_per_dut[namespace].append(interface)
                     downstream_port_ids.append(port_id)
                     downstream_port_id_to_router_mac_map[port_id] = router_mac
                 elif "T3" in neighbor["name"]:
-                    upstream_ports_per_dut[neighbor['namespace']].append(interface)
+                    upstream_ports_per_dut[namespace].append(interface)
                     upstream_port_ids.append(port_id)
                     upstream_port_id_to_router_mac_map[port_id] = router_mac
                 mg_facts = duthost.get_extended_minigraph_facts(tbinfo, namespace)
@@ -420,7 +420,7 @@ def stage(request, duthosts, rand_one_dut_hostname, tbinfo):

 def create_or_remove_acl_table(duthost, acl_table_config, setup, op, topo):
     for sonic_host_or_asic_inst in duthost.get_sonic_host_and_frontend_asic_instance():
-        namespace = sonic_host_or_asic_inst.namespace if hasattr(sonic_host_or_asic_inst, 'namespace') else ''
+        namespace = sonic_host_or_asic_inst.namespace if hasattr(sonic_host_or_asic_inst, 'namespace') else DEFAULT_NAMESPACE
         if op == "add":
             [logger.info](http://logger.info/)("Creating ACL table: \"{}\" in namespace {} on device {}"
                         .format(acl_table_config["table_name"], namespace, duthost))
